File: June/27-06-2024/Toeplitz matrix.java
//{ Driver Code Starts
import java.util.*;

class Check_IsToepliz {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        while (T > 0) {
            int N = sc.nextInt();
            int M = sc.nextInt();
            int arr[][] = new int[N][M];
            for (int i = 0; i < N; i++) {
                for (int j = 0; j < M; j++) arr[i][j] = sc.nextInt();
            }

            Solution g = new Solution();
            boolean b = g.isToeplitz(arr);

            if (b == true)
                System.out.println("true");
            else
                System.out.println("false");

            T--;
        }
    }
}
// } Driver Code Ends


class Solution {
    /*You are required to complete this method*/
    boolean isToeplitz(int mat[][]) {
        // Your code Start
           int r = mat.length;
        int c = mat[0].length;
        
        for (int row = 0; row < r - 1; ++row) {
            for (int col = 0; col < c - 1; ++col) {
                if (mat[row][col] != mat[row + 1][col + 1]) {
                    return false;
                }
            }
        }
        return true;
        //Your Code End
    }
}

File: June/27-06-2024/Toeplitz matrix.js
//{ Driver Code Starts
// Initial Template for javascript

"use strict";

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", (inputStdin) => { inputString += inputStdin; });

process.stdin.on("end", (_) => {
    inputString =
        inputString.trim().split("\n").map((string) => { return string.trim(); });

    main();
});

function readLine() { return inputString[currentLine++]; }

function printArray(arr) {
    let s = "";
    for (let i of arr) {
        s = s + i + " ";
    }
    console.log(s);
}

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for (; i < t; i++) {
        let inputArr = readLine().trim().split(" ").map((x) => parseInt(x));

        let [n, m] = inputArr;
        let A = [];
        inputArr = readLine().trim().split(" ").map((x) => parseInt(x));
        for (let j = 0; j < n * m; j += m) {
            let row = [];
            for (let k = 0; k < m; k++) {
                row.push(inputArr[j + k]);
            }
            A.push(row);
        }
        let obj = new Solution();
        let res = obj.isToeplitz(A);

        console.log(res);
    }
}
// } Driver Code Ends


// User function Template for javascript

/**
 * @param {number[][]} edges
 * @param {number} N
 * @param {number} M
 * @return {boolean}
 */

class Solution {
    isToeplitz(mat) {
        //Your Code Start
        let r = mat.length;
        let c = mat[0].length;
        
        for (let row = 0; row < r - 1; ++row) {
            for (let col = 0; col < c - 1; ++col) {
                if (mat[row][col] !== mat[row + 1][col + 1]) {
                    return false;
                }
            }
        }
        return true;
        //Your Code End
    }
}
